networkit/cpp/graph/GraphBuilder.cpp

```diff
@@ -175,14 +173,34 @@ Graph GraphBuilder::toGraph(bool autoCompleteEdges, bool parallel) {
 }
 
 void GraphBuilder::toGraphDirectSwap(Graph &G) {
-    G.outEdges = std::move(outEdges);
-    G.outEdgeWeights = std::move(outEdgeWeights);
-    G.inEdges = std::move(inEdges);
-    G.inEdgeWeights = std::move(inEdgeWeights);
+    G.removeAllEdges();
```

Comment:
This is a private method that is only called with an empty graph. An assert that the graph is empty would be more appropriate imho.

Reply:
But i don't think the Graph would be empty. DirectSwap we expect the graph to not be empty.

Reply:
`toGraphDirectSwap` is only called in `toGraph` (it is private). There, the graph object is initialized with `Graph G(n, weighted, directed);`. I cannot see any code there that could add edges to the graph that is called before `toGraphDirectSwap`.